homeassistant/components/generic_water_heater/water_heater.py

```diff
@@ -0,0 +1,228 @@
+        self._unit_of_measurement = unit
+        self._current_operation = STATE_ON
+        self._current_temperature = None
+        self._operation_list = [
+            STATE_ON,
+            STATE_OFF,
+        ]
+        self._available = False
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._support_flags
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @property
+    def available(self):
+        """Return if entity is available."""
+        return self._available
+
+    @property
+    def name(self):
+        """Return the name of the water_heater device."""
+        return self._name
+
+    @property
+    def current_temperature(self):
+        """Return current temperature."""
+        return self._current_temperature
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. on, off."""
+        return self._current_operation
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._operation_list
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        self._target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        await self._async_control_heating()
+
+    async def async_set_operation_mode(self, operation_mode):
+        """Set new operation mode."""
+        self._current_operation = operation_mode
+        await self._async_control_heating()
+
+    async def async_added_to_hass(self):
+        """Run when entity about to be added."""
+        await super().async_added_to_hass()
+
+        self.async_on_remove(
+            async_track_state_change_event(
+                self.hass, [self.sensor_entity_id], self._async_sensor_changed
+            )
+        )
+        self.async_on_remove(
+            async_track_state_change_event(
+                self.hass, [self.heater_entity_id], self._async_switch_changed
+            )
+        )
+
+        old_state = await self.async_get_last_state()
+        if old_state is not None:
+            self._target_temperature = float(old_state.attributes.get(ATTR_TEMPERATURE))
+            self._current_operation = old_state.state
+
+        temp_sensor = self.hass.states.get(self.sensor_entity_id)
+        if temp_sensor:
+            self._current_temperature = float(temp_sensor.state)
+
+        heater_switch = self.hass.states.get(self.heater_entity_id)
+        if heater_switch and heater_switch.state not in (
+            STATE_UNAVAILABLE,
+            STATE_UNKNOWN,
+        ):
```

Comment:
Maybe make this tuple a constant since its used in a few places?

Reply:
We are discussing `STATE_UNAVAILABLE, STATE_UNKNOWN` ?

From a readability point of view, I would prefer to leave it